src/adapter/src/catalog.rs

```diff
@@ -2768,6 +2783,16 @@ impl<S: Append> Catalog<S> {
         self.storage().await.allocate_user_id().await
     }
 
+    pub async fn test_only_dont_use_in_production_allocate_system_id(
```

Comment:
`#[cfg(test)]` might work here? I don't know the rust details if it'll be compiled in for things in the `tests` directory or not, but worth a try. :shrug: